src/include/units/concepts.h

```diff
@@ -289,4 +289,21 @@ concept Scalar =
   std::regular<T> &&
   (detail::constructible_from_integral<T> || detail::not_constructible_from_integral<T>);
 
+ template <typename T> 
+ concept Arithmetic = std::is_arithmetic_v<T>;
```

Comment:
As I wrote many times already we cannot constrain the library to fundamental types only. There are a lot of type-safe wrappers around fundamental types that have to work nicely here as well.

Reply:
> As I wrote many times already we cannot constrain the library to fundamental types only. There are a lot of type-safe wrappers around fundamental types that have to work nicely here as well.

There are 3 UDT math types I can find in current standard library with binary math operators

std::complex
std::chrono::duration
std::valarray

**_In each of those types binary operator functions are constrained in both their arguments ( Take a close look at how they are implemented rather than just the documentation). There is no precedent for your statement about unconstrained arguments in the standard library._**

Reply:
Types of https://www.boost.org/doc/libs/release/libs/safe_numerics/ and those being worked on by the WG21's Numerics SG should work as well. Like how `std::chrono::duration`'s `rep` isn't limited to fundamental types.

Reply:
> Types of https://www.boost.org/doc/libs/release/libs/safe_numerics/ and those being worked on by the WG21's Numerics SG should work as well. Like how `std::chrono::duration`'s `rep` isn't limited to fundamental types.


Sure.
We seem to be back to the `Numeric` concept at this point

https://www.boost.org/doc/libs/1_73_0/libs/safe_numerics/doc/html/numeric.html

So Here is an examination of how to do DimensionlessQuantity in a structural way, but allowing customisation ( e.g for the angle case)
 Bear in mind I dont use Concepts much so it can no doubt be done better.

https://godbolt.org/z/Hxfj9K

Anyway it seems to show that it can be made to work if the will is there ;-)

Reply:
Linked here (https://github.com/mpusz/units/issues/27#issuecomment-637359043) is `QuantityRatio`, an interesting test case for DimensionlessQuantity